Userland/Libraries/LibWeb/CSS/Parser/Parser.cpp

```diff
@@ -3329,6 +3330,67 @@ Optional<Color> Parser::parse_rgb_or_hsl_color(StringView function_name, Vector<
     return {};
 }
 
+Optional<EdgeRect> Parser::parse_rect(Vector<ComponentValue> const& component_values)
+{
+    Length params[4];
+    auto tokens = TokenStream { component_values };
+    tokens.skip_whitespace();
+
+    auto maybe_length_1 = parse_length(tokens.next_token().token());
+    if (!maybe_length_1.has_value())
+        return {};
+    params[0] = maybe_length_1.release_value();
+
+    tokens.skip_whitespace();
+    if (tokens.peek_token().is(Token::Type::Comma)) {
+        tokens.next_token();
+    }
+
+    tokens.skip_whitespace();
+    auto maybe_length_2 = parse_length(tokens.next_token().token());
+    if (!maybe_length_2.has_value())
+        return {};
+    params[1] = maybe_length_2.release_value();
+
+    tokens.skip_whitespace();
+    if (tokens.peek_token().is(Token::Type::Comma)) {
+        tokens.next_token();
+    }
```

Comment:
You repeat this almost identical (other than the `params[<index>]`)  block of code multiple times. Maybe just make it a loop and let the compiler handle that :^).

You could do 
```c++
for (auto& param: params) {
  // ...
 param = maybe_length.value(); // set param via the reference.
  // ...
}